src/analyses/constant_propagator.cpp

```diff
@@ -275,31 +312,87 @@ bool constant_propagator_domaint::two_way_propagate_rec(
   std::cout << "two_way_propagate_rec: " << format(expr) << '\n';
 #endif
 
-  bool change=false;
+  bool change = false;
 
   if(expr.id()==ID_and)
   {
     // need a fixed point here to get the most out of it
+    bool change_this_time;
     do
     {
-      change = false;
+      change_this_time = false;
 
       forall_operands(it, expr)
-        if(two_way_propagate_rec(*it, ns, cp))
-          change=true;
+      {
+        change_this_time |= two_way_propagate_rec(*it, ns, cp);
```

Comment:
Would `std::any_of` work?

Reply:
No, as it would terminate early; we want to run propagate on every operand without regard for whether any individual sub-condition propagated anything.